### ISSUE TITLE
Deprecation Warning: str_replace() - Passing Null to Parameter #3 in ee/legacy/libraries/grid_parser.php line 466

### ISSUE THREAD

**Issue by akolinski** [ExpressionEngine/ExpressionEngine#3815]
**Description of the problem**
I encountered a deprecation warning in ExpressionEngine within the ee/legacy/libraries/grid_parser.php file at line 466. The warning is related to the str_replace() function and indicates that null is being passed to parameter #3. This issue should be addressed as it may impact the intended functionality of the str_replace() operation in the specified context.

**How To Reproduce**
Seeing it on the frontend of the website when logged in as an admin with output and debugging turned on.

<img width="1444" alt="Screenshot 2023-10-11 at 11 01 50 am" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/5883593/d8934e9a-bf23-49f5-9370-2d8b2c208d5d">

**Error Messages**

```
Deprecated
str_replace(): Passing null to parameter #2 ($replace) of type array|string is deprecated
ee/legacy/libraries/Grid_parser.php, line 466
Severity: E_DEPRECATED

Warning
Cannot modify header information - headers already sent by (output started at ee/legacy/core/Exceptions.php:120)
ee/ExpressionEngine/Boot/boot.common.php, line 478
Severity: E_WARNING
```

**Environment Details:**
 - Version: 7.3.12
 - PHP Version 8.2
 - MySQL Version 8
 - OS: Linux
 - Web Server: Apache

**Comment by TomJaeger** [comment]
@akolinski Thanks for posting this, as well as the others....  It's a little odd that your running into so many PHP 8.2 issues... any chance you can also pass along a list of add-ons (field types specifically as well) that are installed?

For this grid one specifically can you pass along the fields in this specific grid?  (Super helpful in us replicating) 

Thanks for all the reports... please keep them coming!

**Comment by akolinski** [comment]
@TomJaeger no worries.

List of addons installed on our system.

ACER Assessment quick search 3.0.0
ACER Breadcrumb 3.0.0
ACER Edit this 3.0.1
ACER Region Filter 3.0.0
ACER Resource Router 1.1
ACER SEO 4.0.0
ACER Taxonomy Helper 3.0.0
Acer Utilities 5.1.3
Adman 5.1.8
DataGrab 3.0.3
Block and Allow 4.0.0
Bloqs 5.0.11
Bones 1.1.1
Cache Buster 3.0.0
Custom 404 3.0.0
DB Sessions 1.0.0
Debug override 3.0.0
Detour Pro 3.0.1
Email 2.1.0
Entry Type 4.0.0
Force secure cookies 3.0.0
GWcode categories 3.0.0
HTTP Header 1.0.0
IM2C tools 3.0.0
IP to Nation 3.0.0
JCOGS Image 1.3.6
Last site ID 1.0.0
Low Reorder 4.0.3
Low Replace 3.0.0
Low Seg2Cat 3.0.0
Mustash 3.1.0
Open Search 1.0.0
Pro Search 8.0.2
Publisher 3.10.2
Query 2.0.0
Reading Time Pro 1.0.0
Resource Router 1.1.3
Rich Text Editor 2.1.0
Simple Registration 1.6.3
Site_map 1.0.0
Speedy 1.8.1
Stash 3.2.0
Statistics 2.2.0
Stem submissions 3.0.0
Stringy 5.2.2
Switchee 3.0.1
Tag 7.0.0
Tagger 5.1.0
Taxonomy 3.0.9.1
WFP facilitators 3.0.2
Wygwam 5.4.6

**Comment by akolinski** [comment]
@TomJaeger I took a screenshot of the entry so you can see the different field types of the grids that are used in the channel. 

![Screenshot 2023-10-11 at 11 31 18 am](https://github.com/ExpressionEngine/ExpressionEngine/assets/5883593/0d99f541-8fd9-467d-8451-363009620023)

Let me know of anything else I can help with to help with the debugging.

**Comment by intoeetive** [comment]
@akolinski thank you for the report!

This one will be resolved in upcoming patch release with https://github.com/ExpressionEngine/ExpressionEngine/pull/3782

**Comment by akolinski** [comment]
> @akolinski thank you for the report!
> 
> This one will be resolved in upcoming patch release with #3782

Awesome thanks
